suite_context.go

```diff
@@ -0,0 +1,486 @@
+package godog
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"io"
+	"path/filepath"
+	"reflect"
+	"regexp"
+	"strconv"
+	"strings"
+
+	"github.com/DATA-DOG/godog/gherkin"
+)
+
+// SuiteContext can be used for meta-testing of godog features/steps themselves.
+//
+// A typical user of the godog library should never need this, rather it is
+// provided for those developing add-on libraries for godog.
+//
+// For an example of how to use, see godog's own `features/` and `suite_test.go`.
+func SuiteContext(s *Suite, additionalContextInitializers ...func(suite *Suite)) {
+	c := &suiteContext{
+		extraCIs: additionalContextInitializers,
+	}
+
+	// apply any additional context intializers to modify the context that the
+	// meta-tests will be run in
+	for _, ci := range additionalContextInitializers {
+		ci(s)
+	}
+
+	s.BeforeScenario(c.ResetBeforeEachScenario)
+
+	s.Step(`^a feature path "([^"]*)"$`, c.featurePath)
+	s.Step(`^I parse features$`, c.parseFeatures)
+	s.Step(`^I'm listening to suite events$`, c.iAmListeningToSuiteEvents)
+	s.Step(`^I run feature suite$`, c.iRunFeatureSuite)
+	s.Step(`^I run feature suite with formatter "([^"]*)"$`, c.iRunFeatureSuiteWithFormatter)
+	s.Step(`^a feature "([^"]*)" file:$`, c.aFeatureFile)
+	s.Step(`^the suite should have (passed|failed)$`, c.theSuiteShouldHave)
+
+	s.Step(`^I should have ([\d]+) features? files?:$`, c.iShouldHaveNumFeatureFiles)
+	s.Step(`^I should have ([\d]+) scenarios? registered$`, c.numScenariosRegistered)
+	s.Step(`^there (was|were) ([\d]+) "([^"]*)" events? fired$`, c.thereWereNumEventsFired)
+	s.Step(`^there was event triggered before scenario "([^"]*)"$`, c.thereWasEventTriggeredBeforeScenario)
+	s.Step(`^these events had to be fired for a number of times:$`, c.theseEventsHadToBeFiredForNumberOfTimes)
+
+	s.Step(`^a failing step`, c.aFailingStep)
+	s.Step(`^this step should fail`, c.aFailingStep)
+	s.Step(`^the following steps? should be (passed|failed|skipped|undefined|pending):`, c.followingStepsShouldHave)
+	s.Step(`^all steps should (?:be|have|have been) (passed|failed|skipped|undefined|pending)$`, c.allStepsShouldHave)
+	s.Step(`^the undefined step snippets should be:$`, c.theUndefinedStepSnippetsShouldBe)
+
+	// event stream
+	s.Step(`^the following events should be fired:$`, c.thereShouldBeEventsFired)
+
+	// lt
+	s.Step(`^savybių aplankas "([^"]*)"$`, c.featurePath)
```

Comment:
maybe some of those steps aren't necessary, probably can be extracted later for godog specific test cases. but lets leave it for now just for the sake of simplistic pull request